### PR TITLE
Introduce webpack-dev-server

### PR DESCRIPTION
**Pull request by hyjk2000**
- New dev server supports hot module replacement
- Update Travis-CI config to use new Trusty image
- Prune dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
     "node": "^6.0.0"
   },
   "scripts": {
-    "start": "bin/server",
-    "deploy": "bin/deploy",
-    "build": "node_modules/.bin/webpack"
+    "start": "node_modules/.bin/webpack-dev-server --inline --hot",
```
Comment (hiveer) [inline package.json]: Here we removed this `bin/server`. Does it means we can also remove the file "bin/server"?

Comment (hyjk2000) [inline package.json]: Yeah, your're right. I don't see much use of `bin/server` after this. Nginx should be used as production server if we don't use GitHub Page or CDNs. There's no point to serve static content through node.

File: package.json [hunk 2]
```diff
@@ -19,19 +19,16 @@
...
-  "dependencies": {
-    "bootstrap": "^3.3.7",
-    "jquery": "^3.1.1",
-    "js-cookie": "^2.1.3",
-    "leancloud-storage": "^1.5.3",
-    "requirejs": "^2.3.2"
-  },
   "license": "MIT",
   "devDependencies": {
+    "bootstrap": "^3.3.7",
```
Comment (just4fun) [inline package.json]: Why we need to move all dependencies into devDependencies?

I think `dependencies` is used for production, which means we can run `npm install --save` to install necessary packages in `dependencies` only. During development, if we only run `npm install`, there is no difference.

It's not a big problem, but I'd like to distinguish them in repo.

**Comment by just4fun** [comment]
giphy

File: package.json (hunk 2 above)
Comment (just4fun) [inline package.json]: Since travis will run `npm install` automatically, and we need to use webpack to generate prod files, I think it's OK.

**Pull request merged**